modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/upgrade/v14_0_0/test/ObjectDefinitionUpgradeProcessTest.java

```diff
@@ -0,0 +1,456 @@
+			_objectDefinition2, _objectField.getName(),
+			objectEntry.getObjectEntryId());
+
+		_objectRelationshipLocalService.addObjectRelationshipMappingTableValues(
+			_user.getUserId(), _objectRelationship2.getObjectRelationshipId(),
+			(long)modelAttributes.get("publishedCPDefinitionId"),
+			objectEntry.getObjectEntryId(),
+			ServiceContextTestUtil.getServiceContext());
+
+		Assert.assertTrue(
+			_hasColumn(
+				_objectRelationship2.getDBTableName(), "CPDefinitionId"));
+
+		Assert.assertEquals(
+			1L,
+			_count(
+				"CPDefinitionId",
+				(long)modelAttributes.get("publishedCPDefinitionId"),
+				_objectRelationship2.getDBTableName()));
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME, LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+
+		_objectDefinition1 =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_COMMERCE_PRODUCT_DEFINITION", _company.getCompanyId());
+
+		Assert.assertEquals(
+			"CProductId", _objectDefinition1.getPKObjectFieldDBColumnName());
+		Assert.assertEquals(
+			"CProductId", _objectDefinition1.getPKObjectFieldName());
+
+		_objectDefinition2 = _objectDefinitionLocalService.getObjectDefinition(
+			_objectDefinition2.getObjectDefinitionId());
+
+		_objectField = _objectFieldLocalService.getObjectField(
+			_objectRelationship1.getObjectFieldId2());
+
+		_assertObjectEntryField(
+			(long)modelAttributes.get("CProductId"), _objectDefinition2,
+			_objectField.getName(), objectEntry.getObjectEntryId());
+
+		Assert.assertTrue(
+			_hasColumn(_objectRelationship2.getDBTableName(), "CProductId"));
+
+		Assert.assertEquals(
+			1L,
+			_count(
+				"CProductId", (long)modelAttributes.get("CProductId"),
+				_objectRelationship2.getDBTableName()));
+	}
+
+	private void _assertObjectEntryField(
+			long expectedValue, ObjectDefinition objectDefinition,
+			String objectFieldName, long primaryKey)
+		throws Exception {
+
+		Map<String, Serializable> values =
+			_objectEntryLocalService.
+				getExtensionDynamicObjectDefinitionTableValues(
+					objectDefinition, primaryKey);
+
+		Assert.assertEquals(expectedValue, values.get(objectFieldName));
+	}
+
+	private int _count(String columnName, long primaryKey, String tableName)
+		throws Exception {
+
+		try (Connection connection = DataAccess.getConnection()) {
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"select count(*) from ", tableName, " where ", columnName,
+					" = ?"));
+
+			preparedStatement.setLong(1, primaryKey);
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			if (resultSet.next()) {
+				return resultSet.getInt(1);
+			}
+		}
+
+		return 0;
+	}
+
+	private boolean _hasColumn(String tableName, String columnName)
```

Comment:
@pedro-oliveira446 sort `String tableName, String columnName`

Reply:
I don’t need to make this change because I renamed the method, but this pattern looks correct and matches the other occurrences I found.